gemini/sample-apps/fashion-trends-prediction/app/data_prep/dataProcessing.py

```diff
@@ -0,0 +1,378 @@
+                data=base64.b64decode(encoded_string), mime_type="image/jpeg"
+            )
+        )
+
+
+def generateCaptionGemini(image_path):
+    """Generates a caption for an image using the Gemini model.
+
+    Args:
+                    image_path (str): The path to the image file.
+
+    Returns:
+                    dict: A dictionary containing the generated caption.
+    """
+
+    answer = {}
+
+    with open(image_path, "rb") as image_file:
+        encoded_string = base64.b64encode(image_file.read())
+        user_image = Part.from_data(
+            data=base64.b64decode(encoded_string), mime_type="image/jpeg"
+        )
+
+    prompt = [
+        """
+                You are a fashion editor and your task is to spot fashion trends and extract outfit and fashion items from the image provided .
+                List all outfit items in list of json format, with separate json for each person in the input image.
+                Be as detailed as possible listing shapes, texture, material, length, design pattern, design description and brands.
+                If there is no person in the image, return an empty list and only mention items that are visible in the image.
+                Use fashion terminology if possible and be verbose."""
+        """Input image:""",
+        fewshot_images[0],
+        """Output:
+    [
+        {
+            \"hat\": \"curved brim cotton red yankees baseball cap\",
+            \"sunglasses\": \"black small aviators \",
+            \"jacket\": \"black bomber waist length leather jacket\",
+            \"shirt\": \"white crew neck cotton t-shirt\",
+            \"pants\": \"cotton chinos full length straight leg beige pants\"
+        },
+        {
+            \"jacket\": \"cropped bomber green and white waist height  louis vuitton jacket\",
+            \"shirt\": \"white cotton crop top\",
+            \"skirt\": \"gabardine a-line beige mini skirt\",
+            \"socks\": \"white sheer ankle crew socks\",
+            \"bag\": \"green small rectangular leather top handle shoulder bag\"
+        }
+    ]
+
+
+    Input image:""",
+        fewshot_images[1],
+        """Output:
+[
+        {
+            \"top\": \"brown leopard print mid length cotton top with a square neckline and balloon sleeves and button-up front\",
+            \"jeans\": \"high-waisted,  mid-rise, straight-leg style, light-wash ripped distressed denim jeans\",
+            \"bag\": \"white leather rectangular tote bag\",
+            \"sunglasses\": \"black cat eye style square sunglasses\",
+            \"accessories\": \"choker gold necklace\"
+            \"accessories\": \"choker bangle  bracelet\"
+        }
+]
+
+    Input image:""",
+        user_image,
+        """Output:""",
+    ]
+
+    response = gemini_model.generate_content(
+        prompt,
+        generation_config={
+            "max_output_tokens": 2048,
+            "temperature": 0.4,
+            "top_p": 1,
+            "top_k": 32,
```

Comment:
If you use Gemini 1.5 Pro, you can force the model to output JSON https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/gemini#:~:text=in%20the%20list.-,responseMimeType,-(Preview)